packages/algorithm/src/find.ts

```diff
@@ -132,16 +127,15 @@ export function findIndex<T>(
  * ```
  */
 export function min<T>(
-  object: IterableOrArrayLike<T>,
+  object: Iterable<T>,
   fn: (first: T, second: T) => number
 ): T | undefined {
-  let it = iter(object);
-  let value = it.next();
-  if (value === undefined) {
-    return undefined;
-  }
-  let result = value;
-  while ((value = it.next()) !== undefined) {
```

Comment:
I don't think you're interpreting this correctly. All that the `!== undefined` check did was establish whether the iterator was `done` or not ... but instead of `.done` the Lumino iterators were returning `undefined`.

In the refactored version, the `for ... of` loop will only iterate through items that actually exist. There are no additional calls to `fn(...)` here that weren't happening before.

Reply:
Oh I see. So it was previously impossible for an iterator to yield an `undefined` value, but now it is possible (?). And using it with a fn that returns `number | undefined` isn't really well optimized for, but at least it will return the expected `undefined` value. OK.